### PR TITLE
Jacoco coverage parse implementation && Parse dir

### PR DESCRIPTION
**Pull request by AurelienCaille**
Add a jacoco xml coverage implementation based on a "hello world" jacoco xml report file
Having "/" Parser.input will find xml file in the same dir and parse it, same named files datas will be merge

### CODE REVIEW

File: src/pyinfraboxutils/coverage.py [hunk 1]
```diff
@@ -232,5 +286,29 @@ def __create_markup(self, badge_dir):
         return doc
 
     def parse(self, badge_dir):
-        self.__convert_xml()
+        if(self.input == "/"):
```
Comment (ib-steffen) [inline src/pyinfraboxutils/coverage.py]: this is not correct, the badge_dir is used for something else, this should not be the input directory. you have to check if self.input is a directory or a file.

File: src/pyinfraboxutils/coverage.py [hunk 2]
```diff
@@ -232,5 +286,29 @@ def __create_markup(self, badge_dir):
...
+
+            for f in self.files:
+                if f.name == f1.name:
+                    f1 = f1 + f
+                    self.files.remove(f)
+
+            tmp_files.append(f1)
+
+        self.files = tmp_files
+        self.input = "/"
```
Comment (ib-steffen) [inline src/pyinfraboxutils/coverage.py]: this is probably not needed and could be wrong

File: src/pyinfraboxutils/coverageTest.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python
```
Comment (ib-steffen) [inline src/pyinfraboxutils/coverageTest.py]: please move the test files to `infrabox/test/pyinfraboxutils` and create a Dockerfile which executes the tests. you can probably copy and adapt `infrabox/test/pyinfrabox/Dockerfile`

File: src/pyinfraboxutils/coverage.py (hunk 1 above)
Comment (AurelienCaille) [reply src/pyinfraboxutils/coverage.py]: Ok. Should be fix in next commit

File: src/pyinfraboxutils/coverage.py (hunk 2 above)
Comment (AurelienCaille) [reply src/pyinfraboxutils/coverage.py]: What can wrong with it ? The loop merge File in a temporary list until the self.files is empty and then reverse them. Should I use an other way to merge file ?

File: src/pyinfraboxutils/coverageTest.py (hunk 3 above)
Comment (AurelienCaille) [reply src/pyinfraboxutils/coverageTest.py]: Done in next commit, but not sure the dockerfile is correct

**Pull request merged**